index.bs

```diff
@@ -3631,6 +3635,271 @@ opened</dfn> steps given |window|, |type| and |message|.
+field will be present and consist of an array of integers representing the bytes
+of the cookie value.
+
+#### The network.FetchTimingInfo type #### {#type-network-FetchTimingInfo}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkFetchTimingInfo = {
+    requestTime: uint,
+    redirectStart: uint,
+    redirectEnd: uint,
+    fetchStart: uint,
+    dnsStart: uint,
+    dnsEnd: uint,
+    connectStart: uint,
+    connectEnd: uint,
+    tlsStart: uint,
+    tlsEnd: uint,
+    requestStart: uint,
+    responseStart: uint,
+    responseHeadersEnd: uint,
+    responseEnd: uint,
+};
+</pre>
+
+The <code>NetworkFetchTimingInfo</code> type represents the time of each part of
+the request, relative to <code>requestTime</code>.
+
+TODO: Add service worker fields
+
+#### The network.Header type #### {#type-network-Header}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkHeader = {
+    name: text,
+    ? value: text,
+    ? binaryValue: [ uint ]
+};
+</pre>
+
+The <code>NetworkHeader</code> type represents a single request header.
+
+If the header value can be represented as a UTF-8 encoded string, the
+<code>value</code> field will be present. Otherwise the <code>binaryValue</code>
+field will be present and consist of an array of integers representing the bytes
+of the header.
+
+#### The network.Initiator type #### {#type-network-Initiator}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkInitiator = {
+    type: "parser" / "script" / "preflight" / "other",
+    ?columnNumber: uint,
+    ?lineNumber: uint,
+    ?stackTrace: StackTrace,
+    ?request: NetworkRequest
+};
+</pre>
+
+The <code>NetworkInitiatior</code> type represents the source of a network request.
+
+TODO: Align more closely with Fetch here?
+
+#### The network.Request type #### {#type-network-RequestId}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl remote-cddl">
+NetworkRequest = text;
+</pre>
+
+Each network request has an associated <dfn export>request id</dfn>, which is a string
+uniquely identifying that request.
+
+#### The network.RequestData type #### {#type-network-RequestData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkRequestData = {
+    url: text,
+    method: text,
+    headers: [ *NetworkHeader ],
+    cookies: [ *NetworkCookie ],
+    ?body: text,
+    bodySize: uint,
+    headersSize: uint
+};
+</pre>
+
+The <code>RequestData</code> type represents an ongoing network request.
+
+TODO: Body is actually bytes, not clear how to handle it as text
+
+#### The network.ResponseData type #### {#type-network-ResponseData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkResponseData = {
+    url: text,
+    protocol: text,
```

Comment:
OK, that's what `protocol` is, but CDP only has it in the response. In `beforeRequestSent` I think we might not know the protocol version that will actually be used yet, because even if the application layer says HTTP/1.1 we can negotiate an upgrade to HTTP/2 as part of the initial TLS handshake.

Reply:
Ah ok interesting, in my prototype I mapped the `protocol` here to the `protocolVersion` we have in devtools network data (TLSv1, TLSv1.1, TLSv1.2, TLSv1.3), that's why I was expecting another property. Thanks for the correction

Reply:
I don't _think_ CDP is exposing the low-level TLS details; do we have a use case for also doing that?

Reply:
FWIW, this information is not exposed in HAR, the only usage I know of is the DevTools Network Monitor security tab. I don't think Chrome DevTools show a similar security info for their requests.

Reply:
My proposal is that we consider this out of scope for now; it's easy enough to add `tlsVersion` or something similar later if required.

Reply:
sounds good to me! Regarding the absence of protocol in beforeRequestSent, I don't think this would be a blocker for HAR. We can still pickup the information from the response events corresponding to the same request.